### PR TITLE
Enum improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/576)

### CODE REVIEW

File: downloader/src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -100,18 +98,6 @@ object Main {
...
     }
 
-    private class DataEntityConverter : IStringConverter<DataEntity> {
-        override fun convert(name: String): DataEntity {
-            try {
-                return DataEntity.valueOf(name.toUpperCase())
-            } catch (e: IllegalArgumentException) {
-                e.showStackTrace()
-
-                throw ParameterException("Data entities must be contained in ${DataEntity.ALL}.")
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/Main.kt]: Does JCommander by default show the possible values if an invalid value was provided?
Btw, we need to improve `--help` to list the possible values for all enum parameters.

File: downloader/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -70,14 +70,6 @@ object Main {
     enum class DataEntity {
         PACKAGES,
         PROJECT;
-
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/Main.kt]: Commit message: SHA1 missing after "removed in ...".

Comment (sschuberth) [reply downloader/src/main/kotlin/Main.kt]: Done.

File: downloader/src/main/kotlin/Main.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/Main.kt]: Yes, it does:

    Exception in thread "main" com.beust.jcommander.ParameterException: Invalid value for --entities parameter. Allowed values:[PACKAGES, PROJECT]

Comment (mnonnenmacher) [reply downloader/src/main/kotlin/Main.kt]: Nice.

File: downloader/src/main/kotlin/Main.kt (hunk 2 above)
Comment (mnonnenmacher) [reply downloader/src/main/kotlin/Main.kt]: You shouldn't rebase to not break the link, right? As far as I remember GitHub also rebases when your branch is up-to-date?

Comment (sschuberth) [reply downloader/src/main/kotlin/Main.kt]: True, good point 😞 *Maybe* they changed that by now and do a fast-forward if they can...

Comment (sschuberth) [reply downloader/src/main/kotlin/Main.kt]: Or should I rephrase as "previous commit"?

Comment (sschuberth) [reply downloader/src/main/kotlin/Main.kt]: > *Maybe* they changed that by now and do a fast-forward if they can...

No, they still always rewrite commits 😢 I just checked.

**Pull request merged**